csrc/python_frontend/translation_utils.cpp

```diff
@@ -104,4 +213,64 @@ std::string getString(const BinaryOp* bop) {
+      return "clamp";
+      break;
+    case TernaryOpType::Lerp:
+      return "lerp";
+      break;
+    case TernaryOpType::Threshold:
+      return "threshold";
+      break;
+    case TernaryOpType::Where:
+      return "where";
+      break;
+    default:
+      NVF_CHECK(
+          false,
+          "Unexpected operator type: ",
+          top->getTernaryOpType(),
+          " in ",
+          top->toString());
+  }
+}
+
+#define GET_FUNCTION_TERNARY_SPECIALIZATION(                                 \
+    ResultType, InType1, InType2, InType3)                                   \
+  template <>                                                                \
+  std::function<ResultType(InType1, InType2, InType3)>                       \
+  getFunction<ResultType, InType1, InType2, InType3>(const TernaryOp* top) { \
+    auto wrap_function = [](ResultType (*fn)(InType1, InType2, InType3)) {   \
+      return fn;                                                             \
+    };                                                                       \
+                                                                             \
+    switch (top->getTernaryOpType()) {                                       \
+      case TernaryOpType::Clamp:                                             \
+        return wrap_function(clamp);                                         \
+        break;                                                               \
+      case TernaryOpType::Lerp:                                              \
+        return wrap_function(lerp);                                          \
+        break;                                                               \
+      case TernaryOpType::Threshold:                                         \
+        return wrap_function(threshold);                                     \
+        break;                                                               \
+      case TernaryOpType::Where:                                             \
+        return wrap_function(where);                                         \
+        break;                                                               \
+      default:                                                               \
+        NVF_CHECK(                                                           \
+            false,                                                           \
+            "Unexpected operator type: ",                                    \
+            top->getTernaryOpType(),                                         \
+            " in ",                                                          \
+            top->toString());                                                \
+    }                                                                        \
+  }
+
+// Template specializations for std::function for TernaryOp
+GET_FUNCTION_TERNARY_SPECIALIZATION(TensorView*, TensorView*, Val*, Val*)
```

Comment:
Are the other combinations needed here, like TV,TV,Val and Val,TV,Val etc?

Reply:
These are just for `clamp` & `threshold`. So I don't think we needed anything else.

Reply:
I think moving this closer to where the generic template is would help clearing the confusion.

Reply:
Since the template functions are fully specialized, they must go in the cpp file.